### PR TITLE
Skip interzone diffuse solar exchange factors calculations when no interzone window exists

### LINKED ISSUE #7960: Removed IgnoreInteriorWindowTransmission tag in Output:Diagnostics
Issue overview
--------------
Seems the `IgnoreInteriorWindowTransmission` tag no longer exists in the object `Output:Diagnostics`, but the logic of checking this condition still remains in the [code](https://github.com/NREL/EnergyPlus/blob/274e7df1216d33e62ac53c4e2d75e20600bb83fb/src/EnergyPlus/SimulationManager.cc#L1079
).

Wonder if this logic is meant to be removed as well.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by xuanluo113**
Pull request overview
---------------------
 - Fixes #7960
 - Check if `InterZoneWindow` exists in the model. Skip calculating the diffuse solar exchange factors between zones if not.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by xuanluo113** [comment]
@Myoldmopar Now this one is ready to be reviewed as well. There're do diffs anymore with 0.96% performance improvement.

**Comment by mitchute** [comment]
@xuanluo113 it looks like we still need some sort of a unit test here.

**Comment by xuanluo113** [comment]
> @xuanluo113 it looks like we still need some sort of a unit test here.

@mitchute The unit test was added. Thanks for the review.

File: src/EnergyPlus/SimulationManager.cc [hunk 1]
```diff
@@ -1068,8 +1068,6 @@ namespace SimulationManager {
                     MakeMirroredDetachedShading = false;
                 } else if (UtilityRoutines::SameString(Alphas(NumA), "DoNotMirrorAttachedShading")) {
                     MakeMirroredAttachedShading = false;
-                } else if (UtilityRoutines::SameString(Alphas(NumA), "IgnoreInteriorWindowTransmission")) {
```
Comment (mitchute) [inline src/EnergyPlus/SimulationManager.cc:1071]: Good. This isn't in the IDD, so it makes sense to get rid of it.

Comment (mitchute) [inline src/EnergyPlus/SimulationManager.cc:1071]: Good. This isn't in the IDD, so it makes sense to get rid of it here.

File: src/EnergyPlus/DataReportingFlags.hh [hunk 2]
```diff
@@ -71,7 +71,6 @@ namespace DataReportingFlags {
     extern bool DisplayPerfSimulationFlag;        // True when "Performing Simulation" should be displayed
     extern bool DoWeatherInitReporting;           // Init reporting -- items that go onto OutputFileInits
     extern bool PrintEndDataDictionary;           // Flag for printing "End of Data Dictionary" on output files
-    extern bool IgnoreInteriorWindowTransmission; // True when section "IgnoreInteriorWindowTransmission" is entered
```
Comment (mitchute) [inline src/EnergyPlus/DataReportingFlags.hh:74]: Not used anymore. Good to get rid of it.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 3]
```diff
@@ -3307,9 +3311,15 @@ namespace HeatBalanceSurfaceManager {
 
         // COMPUTE CONVECTIVE GAINS AND ZONE FLUX DENSITY.
         for (int enclosureNum = 1; enclosureNum <= DataViewFactorInformation::NumOfSolarEnclosures; ++enclosureNum) {
-            QS(enclosureNum) *= FractDifShortZtoZ(enclosureNum, enclosureNum) * VMULT(enclosureNum);
-            // CR 8695, VMULT not based on visible
-            QSLights(enclosureNum) *= FractDifShortZtoZ(enclosureNum, enclosureNum) * VMULT(enclosureNum);
+            if (InterZoneWindow) {
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:3314]: I don't fully understand this change but CI seems happy.

File: tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc [hunk 4]
```diff
@@ -2618,4 +2617,333 @@ TEST_F(EnergyPlusFixture, HeatBalanceSurfaceManager_TestInterzoneRadFactorCalc)
 
 }
 
+TEST_F(EnergyPlusFixture, HeatBalanceSurfaceManager_TestInitHBInterzoneWindow)
```
Comment (mitchute) [inline tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc:2620]: Good new test.

**Review by mitchute — APPROVED** [review]
I merged in develop and ran unit tests locally. I also ran regression tests on some files with and without interzone windows. All tests look good.

**Comment by mitchute** [comment]
Everything looks OK here. Merging.

**Pull request merged**